### PR TITLE
EZP-31220: Migrated codebase to Symfony 5

### PR DESCRIPTION
**Pull request by webhdx**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31220](https://jira.ez.no/browse/EZP-31220)
| **Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

#### Doc:
- Twig Intl extension has been dropped (see `twig/extensions` [upgrade doc for IntlExtension](https://github.com/twigphp/Twig-extensions/blob/1fff363f66a49094c47c4710cda272f042bcffc7/README.rst))
- EzPublishMigration Bundle has been dropped.

**TODO**:
- [ ] **Drop TMP commits before merging**
- [x] Migrate codebase to Symfonny 5
- [x] Resolve outstanding Behat issue.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).

### CODE REVIEW

**Comment by emodric** [comment]
Why will you not support Symfony 4.4 LTS?

I realize Symfony 5 is the latest and greatest, but this means that eZ will not have a LTS release of Symfony for another two years which is not ideal.

**Comment by lserwatka** [comment]
We will have LTS sooner. That is why. More info will follow.

**Comment by emodric** [comment]
I'm talking about Symfony LTS, not Symfony. Symfony LTS will be only 5.4 which is due in November 2021.

**Comment by gggeek** [comment]
Otoh Sf 4.4 will be supported until 3rd Q 2023...
What are the planned release date and maintenance window for eZPl 3?

File: eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php [hunk 1]
```diff
@@ -230,6 +230,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $removedVersionsCounter,
             $contentIdsCount
         ));
+        return 0;
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php]: cs-fixer should complain here and in similar places (missing blank line)

File: eZ/Bundle/EzPublishCoreBundle/Features/Context/ConsoleContext.php [hunk 2]
```diff
@@ -10,7 +10,14 @@
 
 class ConsoleContext implements Context
 {
-    use KernelDictionary;
+    /** @var ConfigResolverInterface */
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Features/Context/ConsoleContext.php]: CS: FQCN

File: eZ/Bundle/EzPublishCoreBundle/Features/Context/ConsoleContext.php [hunk 3]
```diff
@@ -20,6 +27,21 @@ class ConsoleContext implements Context
      */
     private $it = [];
 
+    /**
+     * @param ConfigResolverInterface $configResolver
+     * @param string[] $siteaccessList
+     * @param string $defaultSiteaccess
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Features/Context/ConsoleContext.php]: CS
```suggestion
     * @param string[] $siteaccessList
```

File: eZ/Bundle/EzPublishCoreBundle/Resources/config/templating.yml [hunk 4]
```diff
@@ -10,12 +10,6 @@ parameters:
     ezpublish.twig.extension.filesize.suffixes: ['B', 'kB', 'MB', 'GB', 'TB', 'PB', 'EB']
 
 services:
-    # Enable Twig intl extension
-    twig.extension.intl:
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Resources/config/templating.yml]: I think this deserves a BC note in doc

File: eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ConsoleCommandListenerTest.php [hunk 5]
```diff
@@ -31,7 +31,7 @@ class ConsoleCommandListenerTest extends TestCase
     /** @var \eZ\Bundle\EzPublishCoreBundle\EventListener\ConsoleCommandListener */
     private $listener;
 
-    /** @var \Symfony\Component\Console\Input\InputDefinition; */
+    /** @var \Symfony\Component\Console\Input\InputDefinition ; */
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ConsoleCommandListenerTest.php]:
```suggestion
    /** @var \Symfony\Component\Console\Input\InputDefinition */
```

File: eZ/Bundle/PlatformInstallerBundle/src/Command/InstallPlatformCommand.php [hunk 6]
```diff
@@ -104,6 +104,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         if (!$input->getOption('skip-indexing')) {
             $this->indexData($output, $siteaccess);
         }
+        return 0;
```
Comment (alongosz) [inline eZ/Bundle/PlatformInstallerBundle/src/Command/InstallPlatformCommand.php]: CS

File: eZ/Publish/API/Repository/Exceptions/Exception.php [hunk 7]
```diff
@@ -8,11 +8,9 @@
...
 
-use Throwable;
-
 /**
  * Marker interface for all Repository related exceptions.
  */
-interface Exception extends Throwable
+interface Exception extends \Throwable
 {
 }
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Exceptions/Exception.php]: Too much, no changes needed here :P

File: eZ/Publish/API/Repository/Exceptions/NotImplementedException.php [hunk 8]
```diff
@@ -24,7 +22,7 @@ class NotImplementedException extends ForbiddenException
      * @param int $code
      * @param \Exception|null $previous
      */
-    public function __construct($feature, $code = 0, Exception $previous = null)
+    public function __construct($feature, $code = 0, \Exception $previous = null)
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Exceptions/NotImplementedException.php]: Same here

File: eZ/Publish/Core/FieldType/Tests/CountryTest.php [hunk 9]
```diff
@@ -568,8 +568,8 @@ public function provideValidDataForValidate()
      *              ),
      *          )
      *      ),
-     *      array(
-     *          new ValidationError(
+     *          array(
+     *              new ValidationError(
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/Tests/CountryTest.php]: This doesn't add up, it's an example anyway and alignment looked ok (more files affected)

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/FileSizeExtension.php [hunk 10]
```diff
@@ -37,7 +37,7 @@ class FileSizeExtension extends AbstractExtension
     protected $configResolver;
 
     /**
-     * @param  LocaleConverterInterface $localeConverter
+     * @param     LocaleConverterInterface $localeConverter
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/FileSizeExtension.php]: CS: unnecessary change

File: eZ/Publish/Core/Persistence/Legacy/Notification/Gateway.php [hunk 11]
```diff
@@ -35,7 +35,7 @@ abstract class Gateway
      * Update Notification ValueObject in persistent storage.
      * There's no edit feature but it's essential to mark Notification as read.
      *
-     * @param \eZ\Publish\SPI\Persistence\Notification\Notification $notification*
+     * @param \eZ\Publish\SPI\Persistence\Notification\Notification $notification *
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Notification/Gateway.php]:
```suggestion
     * @param \eZ\Publish\SPI\Persistence\Notification\Notification $notification
```

File: eZ/Publish/Core/Search/Legacy/Content/FullTextData.php [hunk 12]
```diff
@@ -46,7 +46,7 @@ class FullTextData extends ValueObject
     /**
      * List of FullTextValue objects corresponding to content object fields (per translation).
      *
-     * @var \eZ\Publish\Core\Search\Legacy\Content\FullTextValue[];
+     * @var \eZ\Publish\Core\Search\Legacy\Content\FullTextValue[] ;
```
Comment (alongosz) [inline eZ/Publish/Core/Search/Legacy/Content/FullTextData.php]:
```suggestion
     * @var \eZ\Publish\Core\Search\Legacy\Content\FullTextValue[]
```

File: eZ/Publish/SPI/FieldType/Tests/FieldTypeTest.php [hunk 13]
```diff
@@ -473,8 +473,8 @@ public function provideValidDataForValidate()
      *              ),
      *          )
      *      ),
-     *      array(
-     *          new ValidationError(
+     *          array(
+     *              new ValidationError(
```
Comment (alongosz) [inline eZ/Publish/SPI/FieldType/Tests/FieldTypeTest.php]: CS, not needed

File: eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/AbstractRandom.php [hunk 14]
```diff
@@ -36,7 +36,7 @@ public function accept(SortClause $sortClause)
      */
     public function applySelect(SelectQuery $query, SortClause $sortClause, $number)
     {
-        /** @var \eZ\Publish\API\Repository\Values\Content\Query\SortClause\Target\RandomTarget $sortClause->targetData */
+        /** @var \eZ\Publish\API\Repository\Values\Content\Query\SortClause\Target\RandomTarget $sortClause ->targetData */
```
Comment (alongosz) [inline eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/AbstractRandom.php]: CS: can be dropped, probably doesn't work anyway.

File: eZ/Publish/Core/Repository/Mapper/ContentDomainMapper.php [hunk 15]
```diff
@@ -171,7 +171,7 @@ public function buildContentProxy(
      * @param string[] $prioritizedLanguages
      * @param bool $useAlwaysAvailable
      *
-     * @return \eZ\Publish\API\Repository\Values\Content\Content[<int>]
+     * @return \eZ\Publish\API\Repository\Values\Content\Content [<int>]
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/Mapper/ContentDomainMapper.php]:
```suggestion
     * @return \eZ\Publish\API\Repository\Values\Content\Content[]
```
anything else is not really supported

File: eZ/Bundle/EzPublishIOBundle/DependencyInjection/Configuration.php [hunk 16]
```diff
@@ -57,7 +57,7 @@ public function getConfigTreeBuilder()
 
     /**
      * @param NodeDefinition $node
-     * @param                $name
+     * @param $name
```
Comment (adamwojs) [inline eZ/Bundle/EzPublishIOBundle/DependencyInjection/Configuration.php]:
```suggestion
     * @param string $name
```
